build.sbt

```diff
@@ -198,12 +206,107 @@ lazy val sdk = crossProject(JVMPlatform, JSPlatform, NativePlatform)
   .crossType(CrossType.Pure)
   .enablePlugins(NoPublishPlugin)
   .in(file("sdk/all"))
-  .dependsOn(`sdk-common`, `sdk-trace`)
+  .dependsOn(core, `sdk-common`, `sdk-trace`)
   .settings(
     name := "otel4s-sdk"
   )
   .settings(scalafixSettings)
 
+//
+// SDK exporter
+//
+
+lazy val `sdk-exporter-common` =
+  crossProject(JVMPlatform, JSPlatform, NativePlatform)
+    .crossType(CrossType.Pure)
+    .in(file("sdk-exporter/common"))
+    .enablePlugins(NoPublishPlugin)
+    .dependsOn(`sdk-common`)
+    .settings(
+      name := "otel4s-sdk-exporter-common",
+      startYear := Some(2023),
+      libraryDependencies ++= Seq(
+        "org.typelevel" %%% "cats-laws" % CatsVersion % Test,
+        "org.typelevel" %%% "discipline-munit" % MUnitDisciplineVersion % Test
+      )
+    )
+    .settings(munitDependencies)
+    .settings(scalafixSettings)
+
+lazy val `sdk-exporter-proto` =
+  crossProject(JVMPlatform, JSPlatform, NativePlatform)
+    .crossType(CrossType.Pure)
+    .enablePlugins(NoPublishPlugin)
+    .in(file("sdk-exporter/proto"))
+    .settings(
+      name := "otel4s-sdk-exporter-proto",
+      Compile / PB.protoSources += baseDirectory.value.getParentFile / "src" / "main" / "protobuf",
+      Compile / PB.targets ++= Seq(
+        scalapb.gen(grpc = false) -> (Compile / sourceManaged).value / "scalapb"
+      ),
+      scalacOptions := {
+        val opts = scalacOptions.value
+        if (tlIsScala3.value) opts.filterNot(_ == "-Wvalue-discard") else opts
```

Comment:
For some reason `-Wconf:src=src_managed/.*:s` does not work in Scala 3.

Reply:
- https://github.com/lampepfl/dotty/pull/18783